### PR TITLE
improve load finished jobs

### PR DESCRIPTION
**Pull request by fviale**
- do not load unnecessary info (task dependencies, task variables, etc)
 - fix workflows tests which were relying on analysing dependencies after the job is finished

Additionally:

 - fix database error when loading results of skipped task
 - improve JobRemoveHandler

### CODE REVIEW

**Review by nebilbenmabrouk — APPROVED** [review]
LGTM

File: scheduler/scheduler-server/src/test/java/functionaltests/workflow/TWorkflowJobs.java [hunk 1]
```diff
@@ -155,6 +158,46 @@ public static JobId testJobSubmission(SchedulerTHelper schedulerHelper, Job jobT
 
         Assert.assertEquals(jInfo.getJobId(), id);
         Assert.assertEquals(JobStatus.RUNNING, jInfo.getStatus());
+        return id;
+    }
+
+    public static JobId waitForJobFinishedAndCompareDependencies(JobId id, SchedulerTHelper schedulerHelper,
```
Comment (mbenguig) [inline scheduler/scheduler-server/src/test/java/functionaltests/workflow/TWorkflowJobs.java:164]: To my pov.. lack of comment in this function (to explain each step, the first "for Task t"  loop and the second one..

**Review by mbenguig — APPROVED** [review]

File: scheduler/scheduler-server/src/test/java/functionaltests/workflow/TWorkflowJobs.java (hunk 1 above)
Comment (fviale) [reply scheduler/scheduler-server/src/test/java/functionaltests/workflow/TWorkflowJobs.java:164]: Indeed, there is an original lack of comment which I did not improve. I will add some explanation

Comment (fviale) [reply scheduler/scheduler-server/src/test/java/functionaltests/workflow/TWorkflowJobs.java:164]: I tried to improve comments, as requested

**Pull request merged**